src/clj/rems/auth/fake_login.clj

```diff
@@ -1,23 +1,38 @@
 (ns rems.auth.fake-login
-  (:require [clojure.string :as str]
-            [compojure.core :refer [GET defroutes]]
+  (:require [compojure.core :refer [GET defroutes]]
             [hiccup.util :refer [url]]
             [rems.layout :as layout]
-            [rems.db.core :as db]
-            [rems.db.users :as users]
+            [rems.auth.oidc :as oidc]
+            [rems.config :refer [env]]
+            [rems.db.test-data-users :refer [+fake-user-data+ +demo-user-data+]]
+            [rems.common.util :refer [assoc-some-in]]
+            [rems.ga4gh :as ga4gh]
             [ring.util.response :refer [redirect]]))
 
+(defn get-fake-login-users []
+  (condp = (:fake-authentication-data env)
+    :test +fake-user-data+
+    :demo +demo-user-data+))
+
 (defn login-url []
   "/fake-login")
 
 (defn logout-url []
   "/fake-logout")
 
 (defn- fake-login [session username]
-  (assoc (redirect "/redirect")
-         :session (assoc session
-                         :identity (users/get-raw-user-attributes username)
-                         :access-token (str "access-token-" username)))) ; fake access token
+  (let [users (get-fake-login-users)
+        id-data (get users username)
+        extra-attributes (select-keys id-data (map (comp keyword :attribute) (:oidc-extra-attributes env)))]
+    (when (oidc/should-map-userid? id-data) (oidc/create-user-mapping! id-data))
+    (-> (redirect "/redirect")
+        (assoc :session session)
+        (assoc-in [:session :access-token] (str "access-token-" username))
+        (assoc-in [:session :identity] (merge {:eppn (or (oidc/get-mapped-userid id-data) (oidc/get-userid id-data))
+                                               :commonName (some id-data [:name :unique_name :family_name :commonName])
+                                               :mail (some id-data [:email :mail])}
+                                              extra-attributes))
+        (assoc-some-in [:session :identity :researcher-status-by] (ga4gh/passport->researcher-status-by id-data)))))
```

Comment:
this feature is mostly similar to OIDC functions but maybe it could also be extracted into common functionality?

Reply:
extracted identity attributes into shared functions, this should be the "beef" in the feature scope